Extension/src/LanguageServer/configurations.ts

```diff
@@ -739,29 +739,52 @@ export class CppProperties {
         const result: string[] = [];
         if (paths) {
             paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
-            return result;
+            if (paths && !glob) {
+                paths.forEach(entry => {
+                    const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+                    result.push(...entries);
+                });
+                return result;
+            }
         }
 
         const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
-            }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
-            // fastGlob silently strip non-found paths. limit that behavior to dynamic paths only
-            const matches: string[] = fastGlob.isDynamicPattern(normalized) ?
-                fastGlob.sync(normalized, { onlyDirectories: true, cwd}) : [res];
-            globResult.push(...matches.map(s => recursive ? s + '**' : s));
+        let recursiveDoubleStar: boolean = false;
```

Comment:
Could you move these variable declarations to where they are initialized in the for loop, to reduce the number of lines?  It doesn't look like these vars are needed outside of the scope of a single for loop iteration.

```
                let recursiveDoubleStar: boolean = res.endsWith('**') && res.endsWith('\\**');
                let recursiveForwardSlash: boolean = res.endsWith('/');
                let recursiveForwardSlashSingleStar: boolean = res.endsWith('/*');
                let recursiveDoubleBackwardSlashDoubleStar: boolean = res.endsWith('\\**');
```

Reply:
Also, if the variables are only needed in the conditional statements, then directly querying `res.endsWith(...)` in the conditional maybe sufficient?